code/API_definitions/qod-api.yaml

```
@@ -352,20 +358,52 @@ components:
         * `QOS_S` - Qualifier for the requested QoS profile _S_
         * `QOS_M` - Qualifier for the requested QoS profile _M_
         * `QOS_L` - Qualifier for the requested QoS profile _L_
+    
     Notification:
       type: object
       required:
         - sessionId
-        - event
+        - eventReports
       properties:
         sessionId:
           $ref: "#/components/schemas/SessionId"
-        event:
-          $ref: "#/components/schemas/SessionEvent"
-    SessionEvent:
+        eventReports:
+          type: array
+          items:
+            $ref: "#/components/schemas/EventReport"
+          minItems: 1
+          description: Contains the reported event and applicable information
+    
+    EventReport:
+      oneOf:
+        - $ref: '#/components/schemas/QosStatusChangedEventReport'
+      discriminator:
+        propertyName: eventType
+        mapping:
+          QOS_STATUS_CHANGED: "#/components/schemas/QosStatusChangedEventReport"
+        
+    QosStatusChangedEventReport:
+      type: object
+      required:
+        - eventType
+        - qosStatus
+      properties:
+        eventType:
+          $ref: "#/components/schemas/SessionEventType"
+        qosStatus:
+          $ref: "#/components/schemas/QosStatus"
+        statusInfo:
```

```suggestion
        statusInfo:
          $ref: "#/components/schemas/StatusInfo"
          
    StatusInfo:
```

As all other properties has its own schema separated